### ISSUE TITLE
Add Jest as test alternative to Mocha

### ISSUE THREAD

**Issue by ericis** [oclif/oclif#173]
**Feature request**

Add Jest as test alternative to Mocha
https://jestjs.io/

**Comment by sourabh1031** [comment]
does it mean i have to add jest to the project

**Comment by ericis** [comment]
@sourabh1031 Your changes look good to me (I am not a project maintainer). Just letting you know.

**Comment by sourabh1031** [comment]
@ericis thanks for your time to review my pr

**Comment by MrStLouis** [comment]
I think this should be able to be closed, but maybe better documented in the testing section. I was able to get this working using the `@oclif/test` package without mocha, although there is still a dependency on chai in `fancy-test`. I removed mocha from my package.json, keeping chai & @types/chai in devDependencies. In my tests, I removed `expect` from @oclif/test and used the global jest expect/describe/test functions. I aliased ocif's `test` function to `cliTest` and used it as documented. no issues running any test.

FWIW, I am using the newest jest (27.5.1) and oclif (^2)

**Comment by jonaslindstr** [comment]
@MrStLouis If you have the time, could you post a small example of a working test case, set up the way you describe? Trying to get this working at the moment, so it would be extremely helpful. 😄

**Comment by jonaslindstr** [comment]
> @MrStLouis If you have the time, could you post a small example of a working test case, set up the way you describe? Trying to get this working at the moment, so it would be extremely helpful. 😄

Never mind, I got it to work with the example project (`hello world`) without even modifying the imports. So the first line is still 
```
import {expect, test} from '@oclif/test'
```
and the tests run with `yarn jest`.

The issue turned out to be that we needed to add the `"files"` parameter to `package.json` due to other source files in the repository causing Jest errors. So it wasn't related to Oclif or Jest, really.
